React InstantSearch/routing-basic/src/App.js

```diff
@@ -25,70 +25,58 @@ const searchStateToUrl = (props, searchState) =>
 
 const urlToSearchState = location => qs.parse(location.search.slice(1));
 
-class App extends Component {
-  state = {
-    searchState: urlToSearchState(this.props.location),
-    lastLocation: this.props.location,
-  };
-
-  static getDerivedStateFromProps(props, state) {
-    if (props.location !== state.lastLocation) {
-      return {
-        searchState: urlToSearchState(props.location),
-        lastLocation: props.location,
-      };
-    }
+export function App({ location, history }) {
+  const [searchState, setSearchState] = useState(urlToSearchState(location));
+  const [debouncedSetState, setDebouncedSetState] = useState(null);
+
+  function onSearchStateChange(updatedSearchState) {
+    clearTimeout(debouncedSetState);
+
+    setDebouncedSetState(
+      setTimeout(() => {
+        history.push(
+          updatedSearchState,
```

Comment:
I'm not very familiar with react-router, but looking at their API ref, `history.push` needs the path to be the first argument here.

Reply:
You're right, I didn't pay much attention because I thought the signatures hadn't changed from the past code (since I only updated the Algolia client).

Addressed in 24293ee.